sklearn/tests/test_common.py

```diff
@@ -294,5 +294,7 @@ def test_search_cv(estimator, check, request):
 @pytest.mark.parametrize("estimator", N_FEATURES_IN_AFTER_FIT_ESTIMATORS,
                          ids=_get_check_estimator_ids)
 def test_check_n_features_in_after_fitting(estimator):
+    if estimator.__class__.__name__ == 'KBinsDiscretizer':
+        estimator.n_bins = 'auto'
```

Comment:
This should be in the ``_set_checking_parameters` function and please add a comment "TODO remove in 1.2" so that we don't forget